src/main/java/com/hazelcast/jdbc/HazelcastJdbcStatement.java

```
@@ -0,0 +1,270 @@
+import java.sql.SQLWarning;
+import java.sql.Statement;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+
+class HazelcastJdbcStatement implements Statement {
+
+    /** Query timeout. */
+    private Duration queryTimeout = Duration.ZERO;
+
+    /** Fetch size. */
+    private int fetchSize;
+
+    /** Whether the statement is closed. */
+    private boolean closed;
+
+    /** Poolable flag. */
+    private boolean poolable;
+
+    ResultSet resultSet;
+
+    private final HazelcastInstance client;
+
+    HazelcastJdbcStatement(HazelcastInstance client) {
+        this.client = client;
+    }
+
+    @Override
+    public ResultSet executeQuery(String sql) throws SQLException {
+        doExecute(sql, Collections.emptyList());
+        return resultSet;
+    }
+
+    @Override
+    public int executeUpdate(String sql) throws SQLException {
+        return 0;
+    }
+
+    @Override
+    public void close() throws SQLException {
+        if (!isClosed()) {
+            closed = true;
+            resultSet.close();
+            resultSet = null;
+        }
+    }
+
+    @Override
+    public int getMaxFieldSize() throws SQLException {
+        return 0;
+    }
+
+    @Override
+    public void setMaxFieldSize(int max) throws SQLException {
+
+    }
+
+    @Override
+    public int getMaxRows() throws SQLException {
+        return 0;
+    }
+
+    @Override
+    public void setMaxRows(int max) throws SQLException {
+    }
+
+    @Override
+    public void setEscapeProcessing(boolean enable) throws SQLException {
+
+    }
+
+    @Override
+    public int getQueryTimeout() throws SQLException {
+        return Math.toIntExact(queryTimeout.getSeconds());
+    }
+
+    @Override
+    public void setQueryTimeout(int seconds) throws SQLException {
+        this.queryTimeout = Duration.ofSeconds(seconds);
+    }
+
+    @Override
+    public void cancel() throws SQLException {
```

Cancel is a very important method but is complicated to implement. Should we throw an unsupported exception for now, instead of ignoring the call?